tests/unit/amo/components/TestSearchForm.js

```diff
@@ -219,6 +219,173 @@ describe(__filename, () => {
+        filters: { query: 'a', addonType: ADDON_TYPE_THEME },
+      }));
+    });
+
+    it('does not fetch suggestions twice on focus when query is present but search has not changed', () => {
+      // This test case reproduces what would happen if user opens
+      // `http://localhost:3000/en-US/firefox/search/?q=aa` and then clicks
+      // into the search bar. It should fetch the suggestions only once.
+      const { store } = dispatchAutocompleteResults({ results: [
+        createFakeAutocompleteResult(),
+        createFakeAutocompleteResult(),
+      ] });
+
+      const { autocomplete: autocompleteState } = store.getState();
+
+      const dispatch = sinon.spy();
+      const wrapper = mountComponent({
+        dispatch,
+        query: 'ad',
+        suggestions: autocompleteState.suggestions,
+      });
+
+      sinon.assert.notCalled(dispatch);
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+
+      // User clicks somewhere else on the UI, triggering the
+      // AUTOCOMPLETE_CANCELLED action.
+      store.dispatch(autocompleteCancel());
+      wrapper.setProps(mapStateToProps(store.getState()));
+
+      // User focuses the search input.
+      wrapper.find('input').simulate('focus');
+
+      // User clicks somewhere else on the UI, triggering the
+      // AUTOCOMPLETE_CANCELLED action, again.
+      store.dispatch(autocompleteCancel());
+      wrapper.setProps(mapStateToProps(store.getState()));
+
+      // User focuses the search input, again.
+      wrapper.find('input').simulate('focus');
+
+      expect(wrapper.find(Suggestion)).toHaveLength(
+        autocompleteState.suggestions.length
+      );
+
+      sinon.assert.callCount(dispatch, 1);
+      sinon.assert.calledWith(dispatch, autocompleteStart({
+        errorHandlerId: errorHandler.id,
+        filters: { query: 'ad' },
+      }));
+    });
+
+    it('fetches suggestions on focus when query is present but search has changed', () => {
+      const { store } = dispatchAutocompleteResults({ results: [
+        createFakeAutocompleteResult(),
+        createFakeAutocompleteResult(),
+      ] });
+
+      const { autocomplete: autocompleteState } = store.getState();
+
+      const dispatch = sinon.spy();
+      const wrapper = mountComponent({
+        dispatch,
+        query: 'ad',
+        suggestions: autocompleteState.suggestions,
+      });
+
+      sinon.assert.notCalled(dispatch);
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+
+      // User focuses the search input, which will load suggestions from the API.
+      wrapper.find('input').simulate('focus');
+
+      // User types 'adb' in the search input, which requires new suggestions
+      // to be loaded from the API.
+      wrapper.find('input').simulate('change', createFakeChangeEvent('adb'));
+
+      expect(wrapper.find(Suggestion)).toHaveLength(
+        autocompleteState.suggestions.length
+      );
+
+      sinon.assert.callCount(dispatch, 2);
```

Comment:
Same here; I think doing the check before `wrapper.find('input').simulate('change', createFakeChangeEvent('adb'));` and then doing `dispatch.reset()` then checking again would be a bit more robust and also more clearly show which assertion is related to specific actions we're simulating in the test.